pkg/v1/tkg/tkgctl/create_cluster_test.go

```diff
@@ -327,7 +436,218 @@ var _ = Describe("Unit tests for - ccluster.yaml as input file for 'tanzu cluste
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereControlPlaneEndpoint)
+			Expect("http://10.0.200.101").To(Equal(fmt.Sprintf("%v", mappedVal)))
+			// check that cluster options also upated
+			Expect("http://10.0.200.101").To(Equal(options.VsphereControlPlaneEndpoint))
+
+			// check value for "spec.topology.variables.controlPlane.network.nameservers"
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableControlPlaneNodeNameservers)
+			Expect("100.64.0.0").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			//  check value for "spec.topology.variables.controlPlane.machine.numCPUs
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereCPNumCpus)
+			Expect("2").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			//  check value for "spec.topology.variables.worker.machine.memoryMiB" ConfigVariableVsphereWorkerMemMib
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereWorkerMemMib)
+			Expect("16384").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			// check value for "spec.topology.workers.machineDeployments.2.failureDomain"  - ConfigVariableVsphereAz2
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereAz2)
+			Expect("us-east-1c").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+		})
+	})
+})
+
+var _ = Describe("Unit tests for - (Azure) - cluster_azure.yaml as input file for 'tanzu cluster create -f cluster_azure.yaml' use case", func() {
+	var (
+		ctl       tkgctl
+		tkgClient = &fakes.Client{}
+		bomClient = &fakes.TKGConfigBomClient{}
+		options   CreateClusterOptions
+	)
+	JustBeforeEach(func() {
+		tkgConfigReaderWriter, _ := tkgconfigreaderwriter.NewReaderWriterFromConfigFile(configFilePath, configFilePath)
+		ctl = tkgctl{
+			configDir:              testingDir,
+			tkgClient:              tkgClient,
+			tkgConfigReaderWriter:  tkgConfigReaderWriter,
+			tkgConfigUpdaterClient: tkgconfigupdater.New(testingDir, nil, tkgConfigReaderWriter),
+			tkgBomClient:           bomClient,
+		}
+		tkgClient.IsFeatureActivatedReturns(true)
+	})
+	Context("'tanzu cluster create -f cluster_azure.yaml' use case - vsphere provider - devcc plan:", func() {
+		BeforeEach(func() {
+			options = CreateClusterOptions{
+				ClusterName:            "test-cluster",
+				Plan:                   "dev",
+				InfrastructureProvider: "",
+				Namespace:              "",
+				GenerateOnly:           false,
+				TkrVersion:             fakeTKRVersion,
+				SkipPrompt:             true,
+				Edition:                "tkg",
+				ClusterConfigFile:      classInputFileAzure,
+			}
+		})
+		It("'tanzu cluster create -f cluster_azure.yaml' use case - vsphere provider - devcc plan:, make sure legacy configurations are updated..", func() {
+
+			// Process input cluster yaml file, this should process input cluster yaml file
```

Comment:
Can the first few dozen lines of the test before factored out. They look identical in all 3 cases

Reply:
The first section of lines in each use case (aws, azure, Vsphere) the logic is the same, we need to process the input Cluster Class file that's why those lines are the same except for the input file name.